src/app/dao/tests/PartDAO.spec.ts

```diff
@@ -898,154 +710,300 @@ describe('PartDAO', () => {
+
+                // Then it should fail
+                await expectPermissionToBeDenied(result);
             });
+            describe('moves', () => {
+                it('should allow creating a move on player turn', async() => {
+                    // Given an ongoing part
+                    const partId: string = await setupStartedPartAsPlayerZero();
 
-        });
-        describe('increasing time', () => {
-            it('should allow increasing the time of the opponent (as playerZero)', async() => {
-                await checkTimeUpdate(Player.ZERO,
-                                      (zero: number, one: number): Partial<Part> => {
-                                          return { remainingMsForOne: one + 1000 };
-                                      }, true);
+                    // When creating a move
+                    const result: Promise<string> = gameEventService.addMove(partId, Player.ZERO, { x: 0, y: 0 });
+
+                    // Then it should succeed
+                    await expectAsync(result).toBeResolved();
+                });
+                it('should forbid creating a move on opponent turn', async() => {
+                    // Given an ongoing part where it is the opponent's turn
+                    const partId: string = await setupStartedPartAsPlayerZero();
+                    await gameEventService.addMove(partId, Player.ZERO, { x: 0, y: 0 });
+                    await partDAO.update(partId, { turn: 1 });
+
+                    // When creating a move during the opponent's turn
+                    const result: Promise<string> = gameEventService.addMove(partId, Player.ZERO, { x: 0, y: 0 });
+
+                    // Then it should fail
+                    await expectPermissionToBeDenied(result);
+                });
             });
-            xit('should forbid increasing its own time (as playerZero)', async() => {
-                // Disabled because we removed it from the rules, otherwise
-                // security rules are too complexed to be checked
-                await checkTimeUpdate(Player.ZERO,
-                                      (zero: number, one: number): Partial<Part> => {
-                                          return { remainingMsForZero: zero + 1000 };
-                                      }, false);
+            describe('actions', () => {
+                it('should allow creating AddTurnTime action', async() => {
+                    // Given an ongoing part
+                    const partId: string = await setupStartedPartAsPlayerZero();
+
+                    // When creating an AddTurnTime action
+                    const result: Promise<string> = gameEventService.addAction(partId, Player.ZERO, 'AddTurnTime');
+
+                    // Then it should succeed
+                    await expectAsync(result).toBeResolved();
+                });
+                it('should allow creating AddGlobalTime action', async() => {
```

Comment:
y a un "and forbid creating it as the opponent" ?

Reply:
... maintenant oui !